pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AccessorMethodGenerationRule.java

```diff
@@ -22,6 +30,8 @@
 
 public class AccessorMethodGenerationRule extends AbstractJavaRule {
 
+    private Map<String, Boolean> cache = new HashMap<>();
```

Comment:
override the `end` method of `AbstractJavaRule` and have it call `cache.clear()`, so we don't leak memory after analyzing each file.